drafthorse/pdf.py

```diff
@@ -38,10 +37,41 @@
     create_string_object,
 )
 
+from drafthorse.xmp_schema import XMP_SCHEMA
+
+logging.basicConfig()
 logger = logging.getLogger("drafthorse")
+logger.setLevel(logging.INFO)
 
 
-def attach_xml(original_pdf, xml_data, level="BASIC"):
+def attach_xml(original_pdf, xml_data, level=None, metadata=None, lang=None):
+    """
+    Create the ZUGFeRD invoice by attaching
+    the input XML and proper metadata
+    :param original_pdf: Input PDF
+    :param xml_data: Input XML
+    :param level: optional Factur-X profile level
+    one of {'MINIMUM', 'BASIC WL', 'BASIC', 'EN 16931', 'EXTENDED', 'XRECHNUNG'}
+    if omitted autodetection is performed
+    :type level: string
+    :param metadata: optional dict with user defined PDF metadata
+    for fields "author", "keywords", "title" and "subject". If metadata is None (default value),
+    this lib will generate some metadata by extracting relevant info from the Factur-X/Order-X XML.
+    Here is an example for the metadata argument:
+    pdf_metadata = {
+        'author': 'MyCompany',
+        'keywords': 'Factur-X, Invoice',
+        'title': 'MyCompany: Invoice I1242',
+        'subject':
+          'Factur-X invoice I1242 dated 2017-08-17 issued by MyCompany',
+    }
+    :type metadata: dict
+    :param lang: Language identifier in RFC 3066 format to specify the
+    natural language of the PDF document. Used by PDF readers for blind people.
+    Example: en-US or fr-FR
+    :type lang: string
+    :return: Output PDF containing the metadata and XML
+    """
```

Comment:
```suggestion
    :param level: optional Factur-X profile level
    one of ``{'MINIMUM', 'BASIC WL', 'BASIC', 'EN 16931', 'EXTENDED', 'XRECHNUNG'}``.
    If omitted, autodetection is performed
    :type level: string
    :param metadata: optional dict with user defined PDF metadata
    for fields "author", "keywords", "title" and "subject". If metadata is None (default value),
    this lib will generate some metadata by extracting relevant info from the Factur-X/Order-X XML.
    Here is an example for the metadata argument:
    ```
    pdf_metadata = {
        'author': 'MyCompany',
        'keywords': 'Factur-X, Invoice',
        'title': 'MyCompany: Invoice I1242',
        'subject':
          'Factur-X invoice I1242 dated 2017-08-17 issued by MyCompany',
    }
    ```
    :type metadata: dict
    :param lang: Language identifier in RFC 3066 format to specify the
    natural language of the PDF document. Used by PDF readers for blind people.
    Example: en-US or fr-FR
    :type lang: string
    :return: Output PDF containing the metadata and XML
    """
```